measure_extinction/merge_obsspec.py

```diff
@@ -254,38 +254,46 @@ def merge_spex_obsspec(obstables, output_resolution=2000):
     output_table : astropy Table object
         merged spectra
     """
-    wave_range = [0.8, 5.5] * u.micron
-
+    waves = obstable["WAVELENGTH"].data * 1e4
+    fluxes = obstable["FLUX"].data
+    uncs = obstable["ERROR"].data
+    npts = np.full((len(obstable["FLUX"])), 1.0)
+    npts[obstable["FLAG"] == 1.0] = 0.0
+
+    # determine the wavelength range and calculate the wavelength grid
+    if np.max(waves) < 25000:
+        wave_range = [0.8, 2.45] * u.micron
+    elif np.min(waves) > 24000:
+        wave_range = [2.4, 5.5] * u.micron
+    else:
+        wave_range = [0.8, 5.5] * u.micron
```

Comment:
The SpeX wavelength range should be fixed regardless of the wavelength of the data being used.  By fixing the wavelength range to always be the same, then the rest of the code does not have to do any extrapolation/interpolation, the extinction curve can just be computed for the full wavelength range.  As any part of the wavelength range that does not have data will get npts=0, it is easy to filter out points with no data.

My idea here is to have the SXD and LXD input to this function via a list of astropy tables.  Then the merge could should scale and then merge the two spectra into a single one.

Reply:
Done!